sunpy/database/database.py

```diff
@@ -389,7 +398,27 @@ def _download_and_collect_entries(self, query_result, **kwargs):
         if client is None:
             client = VSOClient()
 
-        paths = client.get(query_result, path, methods).wait(progress=progress)
+        remove_list = []
+        delete_entries = []
+        for qr in query_result:
+            temp = tables.DatabaseEntry._from_query_result_block(qr)
+            for database_entry in self:
+                if database_entry.path is not None and temp._compare_attributes(
+                    database_entry, ["source", "provider", "physobs", "fileid",
+                                     "observation_time_start", "observation_time_end",
+                                     "instrument", "size", "wavemin", "wavemax"]):
+                    if overwrite is False:
```

Comment:
@Cadair I did it to avoid code repetition. If i keep a list of matched_entries, then there will be some lines repeated inside the 2 if conditions for overwrite = True and overwrite = False